lib/weka/core/instances.rb

```diff
@@ -62,9 +62,30 @@ def add_attributes(&block)
         self
       end
 
-      alias with_attributes  add_attributes
-      alias instances_count  num_instances
-      alias attributes_count num_attributes
+      alias with_attributes         add_attributes
+      alias instances_count         num_instances
+      alias attributes_count        num_attributes
+      alias has_string_attribute?   check_for_string_attributes
+
+      ## Check if the instances has any attribute of the given type
+      # @param [String, Symbol, Integer] type type of the attribute to check
+      #   String and Symbol argument are converted to corresponding type
+      #   defined in Weka::Core::Attribute
+      #
+      # @example Passing String
+      #   instances.has_attribute_type?('string')
+      #   instances.has_attribute_type?('String')
+      #
+      # @example Passing Symbol
+      #   instances.has_attribute_type?(:String)
+      #
+      # @example Passing Integer
+      #   instances.has_attribute_type?(Attribute::STRING)
+      def has_attribute_type?(type)
+        type = map_attribute_type(type) unless type.is_a? Integer
+        return false if type.nil?
+        check_for_attribute_type(type)
+      end
```

Comment:
I see, right. Then we could just return -1 (never existing interger index for attributes) instead of nil in `map_attribute_type` and the `return false if type.nil?` line can be removed.